### PR TITLE
build: add emil_build_for macro

### PR DESCRIPTION
**Pull request by richardapeters**

### CODE REVIEW

File: cmake/emil_build_for.cmake [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# 'emil_build_for' uses the EXCLUDE_FROM_ALL property on targets to specify whether the
+# target should be built for a certain combination of properties.
+#
+# These properties are supported:
+#
+# HOST (values: All, Windows, Linux, Darwin): Build the target when building for any all of the host OSes
+# TARGET_MCU_VENDOR: Build the target when building for this vendor
+# TARGET_MCU_FAMILY: Build the target when building for this MCU family
+# TARGET_MCU: Build the target when building for this MCU
+# TARGET_MCU VARIANT: Build the target when building for this MCU variant
```
Comment (rjaegers) [inline cmake/emil_build_for.cmake:10]:
```suggestion
# TARGET_MCU_VARIANT: Build the target when building for this MCU variant
```

File: cmake/emil_build_for.cmake [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+# BOOL: Build the target when this boolean value evaluates to true
+#
+# PREREQUISITE_BOOL: Build the target only when this boolean value evaluates to true
+# PREREQUISITE_CONFIG: Build the target only when building for this configuration
+#
+# The HOST, TARGET_*, and BOOL properties are ORed together. Its result is ANDed with the PREREQUISITE_* properties.
+# Specifying PREREQUISITE_CONFIG results in a generator expression being ued for EXCLUDE_FROM_ALL
```
Comment (rjaegers) [inline cmake/emil_build_for.cmake:17]:
```suggestion
# Specifying PREREQUISITE_CONFIG results in a generator expression being used for EXCLUDE_FROM_ALL
```

File: cmake/emil_build_for.cmake [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# 'emil_build_for' uses the EXCLUDE_FROM_ALL property on targets to specify whether the
+# target should be built for a certain combination of properties.
+#
+# These properties are supported:
```
Comment (rjaegers) [inline cmake/emil_build_for.cmake:4]: Should we describe that all properties are optional?

I.e. that you could also use the function like this:
`emil_buid_for(target HOST Linux)`

**Review by rjaegers — APPROVED** [review]

File: cmake/emil_build_for.cmake (hunk 3 above)
Comment (richardapeters) [reply cmake/emil_build_for.cmake:4]: I'll add that as an  example

**Pull request merged**